tests/jax/core_system_components/executor_test.py

```diff
@@ -13,266 +13,234 @@
+    observation = np.array([1, 2, 3, 4])
+    state = np.array([5, 6, 7, 8])
 
-@pytest.fixture
-def test_system_except_trainer() -> System:
-    """Add description here."""
-    return TestSystemExceptTrainer()
+    # Manually load info into store
+    test_executor.store.action_info = "action_info"
+    test_executor.store.policy_info = "policy_info"
 
+    assert test_executor.select_action(
+        agent=agent, observation=observation, state=state
+    ) == ("action_info", "policy_info")
 
-# TODO: fix test.
-@pytest.mark.skip(reason="test is currently breaking ci pipeline")
-def test_except_trainer(
-    test_system_except_trainer: System,
-) -> None:
-    """Test if the parameter server instantiates processes as expected."""
+    assert test_executor.store.agent == agent
+    assert (test_executor.store.observation == observation).all()
+    assert (test_executor.store.state == state).all()
 
-    # Environment.
-    environment_factory = functools.partial(
-        debugging_utils.make_environment,
-        env_name="simple_spread",
-        action_space="discrete",
-    )
 
-    # Networks.
-    network_factory = mappo.make_default_networks
-
-    # Build the system
-    test_system_except_trainer.build(
-        environment_factory=environment_factory,
-        network_factory=network_factory,
-        executor_parameter_update_period=20,
-        multi_process=False,
-        run_evaluator=True,
-        num_executors=1,
-        use_next_extras=False,
+def test_select_actions_store(
+    test_executor: Executor,
+) -> None:
+    """Test that store is handled properly in select_actions"""
+    observations = {
+        "agent_0": np.array([1, 2, 3, 4]),
+        "agent_1": np.array([5, 6, 7, 8]),
+    }
+
+    # Manually load info into store
+    actions_info = {"agent_0": "actions_info_0", "agent_1": "actions_info_1"}
+    policies_info = {"agent_0": "policies_info_0", "agent_1": "policies_info_1"}
+
+    test_executor.store.actions_info = actions_info
+    test_executor.store.policies_info = policies_info
+
+    assert test_executor.select_actions(observations=observations) == (
+        actions_info,
+        policies_info,
     )
 
-    (
-        data_server,
-        parameter_server,
-        executor,
-        evaluator,
-        trainer,
-    ) = test_system_except_trainer._builder.store.system_build
+    assert test_executor.store.observations == observations
 
-    assert isinstance(executor, acme.core.Worker)
 
-    # Step the executor
-    executor.run_episode()
+def test_update_store(
+    test_executor: Executor,
+) -> None:
+    """Test that store is handled properly in update"""
+    test_executor.update(wait=False)
+    assert not test_executor.store._wait
+    test_executor.update(wait=True)
+    assert test_executor.store._wait
+
+
+def test_init_hook_order(test_executor: TestExecutor) -> None:
+    """Test if init hooks are called in the correct order"""
+    assert test_executor.hook_list == [
```

Comment:
This is great, nice that we are testing hook order. We should also do this in other classes.

Reply:
It really is a nice way to go about testing the core components. We have been doing this for all core components where the hooks get called sequentially.